internal-packages/ai/scripts/generate-schemas.ts

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env tsx
+
+/**
+ * Build script to generate static JSON schemas from tool definitions
+ * This extracts schemas at build time so they can be imported client-side
+ * without pulling in server dependencies
+ */
+
+import fs from 'fs';
+import path from 'path';
+import { createHash } from 'crypto';
+import { zodToJsonSchema } from 'zod-to-json-schema';
+
+// Import all tools
+import checkSpellingGrammarTool from '../src/tools/check-spelling-grammar';
+import extractFactualClaimsTool from '../src/tools/extract-factual-claims';
+import factCheckerTool from '../src/tools/fact-checker';
+import checkMathWithMathJsTool from '../src/tools/check-math-with-mathjs';
+import checkMathTool from '../src/tools/check-math';
+import extractMathExpressionsTool from '../src/tools/extract-math-expressions';
+import extractForecastingClaimsTool from '../src/tools/extract-forecasting-claims';
+import documentChunkerTool from '../src/tools/document-chunker';
+import fuzzyTextLocatorTool from '../src/tools/fuzzy-text-locator';
+import { detectLanguageConventionTool } from '../src/tools/detect-language-convention';
+
+const tools = {
+  'check-spelling-grammar': checkSpellingGrammarTool,
+  'extract-factual-claims': extractFactualClaimsTool,
+  'fact-checker': factCheckerTool,
+  'check-math-with-mathjs': checkMathWithMathJsTool,
+  'check-math': checkMathTool,
+  'extract-math-expressions': extractMathExpressionsTool,
+  'extract-forecasting-claims': extractForecastingClaimsTool,
+  'document-chunker': documentChunkerTool,
+  'fuzzy-text-locator': fuzzyTextLocatorTool,
+  'detect-language-convention': detectLanguageConventionTool,
```

Comment:
I imagine we should have this dictionary stored somewhere else in the tools directory or something.